internal/ethapi/transaction_args.go

```diff
@@ -289,6 +289,19 @@ func (args *TransactionArgs) toTransaction() *types.Transaction {
 				Data:                args.data(),
 				AccessList:          al,
 			}
+		} else if isGingerbreadP2 && args.FeeCurrency != nil {
```

Comment:
Strictly speaking, is not the same case because if it is gingerbread p2 and has a feeCurrency AND something as the gatewayFee, it will create a Tx v2 without the gatewayFee, which will make a valid and executable tx, instead of something that is suppose to fail.
From my point of view, I prefer that the user knows that is has an error but knowing that the transaction that is going to execute is exactly the one that it wants.

We can have or
```
if args.GatewayFeeRecipient != nil || args.GatewayFee != nil || (!isGingerbreadP2 && args.FeeCurrency != nil) {
    [...CeloDynamicV1...]
} else if --isGingerbreadP2 &&--  args.FeeCurrency != nil { // We can remove the isGingerbreadP2 Check otherwise it would have use the last one but I thought it was going to be clearer
    [...CeloDynamicV2...]
} else {
    [...Dynamic...]
}
```
or
```
if args.FeeCurrency == nil && args.GatewayFeeRecipient == nil && args.GatewayFee == nil {
   [...Dynamic...]
} else if isGingerbreadP2 && args.FeeCurrency == nil && args.GatewayFeeRecipient == nil {
   [...CeloDynamicV1...]
} else {
   [...CeloDynamicV2...]
}
```
But to be honest everything is ugly, haha

Reply:
I agree, let keep it.